web/webkit/src/main/scala/net/liftweb/http/SecurityRules.scala

```diff
@@ -0,0 +1,461 @@
+              JField("originalPolicy", content)
+          }
+          violationJson = camelCasedJson \ "csp-report"
+          extractedViolation <- tryo(violationJson.extract[ContentSecurityPolicyViolation])
+        } yield {
+          extractedViolation
+        }
+
+      () => {
+        violation match {
+          case Full(violation) =>
+            LiftRules.contentSecurityPolicyViolationReport(violation) or
+              Full(OkResponse())
+
+          case _ =>
+            logger.warn(
+              s"Got a content security violation report we couldn't interpret: '${request.body.map(new String(_, "UTF-8"))}'."
+            )
+
+            Full(BadRequest("Unrecognized format for content security policy report."))
+        }
+      }
+  }
+}
+
+/**
+ * Defines restrictions on allowing served pages to be embedded in frames.
+ */
+sealed trait FrameRestrictions {
+  def headers: List[(String,String)]
+
+  /**
+   * Returns the headers implied by these frame restrictions.
+   *
+   * Because of how frame restrictions are handled, if enforcement in dev mode
+   * is turned off, no headers are generated in dev mode.
+   */
+  def headers(enforceInDevMode: Boolean = false): List[(String,String)] = {
+    if (! enforceInDevMode && Props.devMode) {
+      Nil
+    } else {
+      headers
+    }
+  }
+}
+object FrameRestrictions {
+  /**
+   * Allows other pages from the same origin as the one being served to embed
+   * this page in a frame.
+   */
+  case object SameOrigin extends FrameRestrictions {
+    val headers = List("X-Frame-Options" -> "SAMEORIGIN")
+  }
+  /**
+   * Does not allow embedding the page being served in a frame at all.
+   */
+  case object Deny extends FrameRestrictions {
+    val headers = List("X-Frame-Options" -> "DENY")
+  }
+}
+
+/**
+ * Specifies security rules for a Lift application. By default, HTTPS is not
+ * required and `Content-Security-Policy` is restricted to the current domain
+ * for everything except images, which are accepted from any domain.
+ * Additionally, served pages can only be embedded in other frames from
+ * the current domain.
+ *
+ * You can use `[[SecurityRules.secure]]` to enable more restrictive, but
+ * also more secure, defaults.
+ *
+ * @param enforceInDevMode If true, security policies and HTTPS rules are
+ *        enforced in dev mode in addition to staging/pilot/production/etc.
+ * @param logInDevMode If true, dev mode violations of security policies are
+ *        logged by default. Note that if you override
+ *        `[[LiftRules.contentSecurityPolicyViolationReport]]` or otherwise
+ *        change the default Lift policy violation handling behavior, it will
+ *        be up to you to handle this property as desired.
+ */
+final case class SecurityRules(
+  https: Option[HttpsRules] = None,
+  content: Option[ContentSecurityPolicy] = Some(ContentSecurityPolicy()),
+  frameRestrictions: Option[FrameRestrictions] = Some(FrameRestrictions.SameOrigin),
```

Comment:
Yes, they give us proper naming and enumeration of the available kinds of security rules. Also, their purpose is not solely providing headers; for example, we currently check early in request servicing if https rules are enabled but set to log-only for dev mode, and log a warning when we get a non-https request. This log-only behavior is not supported by the HTTP header.